### PR TITLE
HIVE-23789: Merge ValidTxnManager into DriverTxnHandler (Miklos Gergely)

### PR DESCRIPTION
**Pull request by miklosgergely**

### CODE REVIEW

File: ql/src/java/org/apache/hadoop/hive/ql/Compiler.java [hunk 1]
```diff
@@ -188,7 +188,6 @@ private BaseSemanticAnalyzer analyze() throws Exception {
     // because at that point we need access to the objects.
     Hive.get().getMSC().flushCache();
 
-    driverContext.setBackupContext(new Context(context));
```
Comment (pvary) [inline ql/src/java/org/apache/hadoop/hive/ql/Compiler.java:191]: Are we sure about this?
We create a backup context so if we have to reexecute the query then we have a context at hand (for removing temporary files etc)

File: ql/src/java/org/apache/hadoop/hive/ql/DriverTxnHandler.java [hunk 2]
```diff
@@ -288,15 +313,231 @@ private void acquireLocksInternal() throws CommandProcessorException, LockExcept
...
+  }
+
+  private List<String> getTransactionalTables(Map<String, Table> tables) {
+    return tables.entrySet().stream()
+      .filter(entry -> AcidUtils.isTransactionalTable(entry.getValue()))
+      .map(Map.Entry::getKey)
+      .collect(Collectors.toList());
+  }
+
+  void addHiveLocksFromContext() {
```
Comment (pvary) [inline ql/src/java/org/apache/hadoop/hive/ql/DriverTxnHandler.java:532]: I do not like storing lock related stuff in context. Shouldn't this be private to the DbTxnHandler?

File: ql/src/java/org/apache/hadoop/hive/ql/Compiler.java (hunk 1 above)
Comment (miklosgergely) [reply ql/src/java/org/apache/hadoop/hive/ql/Compiler.java:191]: The usage of backupContext was removed by Peter Varga recently (https://github.com/apache/hive/commit/e2a02f1b43cba657d4d1c16ead091072be5fe834#diff-71a166c053d9c698f9cb64eaef832aff), I've asked him to confirm, and it was intentional. After this change we are only setting the backup context here, but it is never used.

File: ql/src/java/org/apache/hadoop/hive/ql/DriverTxnHandler.java (hunk 2 above)
Comment (miklosgergely) [reply ql/src/java/org/apache/hadoop/hive/ql/DriverTxnHandler.java:532]: I agree, but in a separate jira. This one is for merging the two classes only, so we'll have a clean and easy to understand history.

**Pull request merged**